fineract-core/src/main/java/org/apache/fineract/infrastructure/core/service/database/DatabaseSpecificSQLGenerator.java

```diff
@@ -36,14 +42,26 @@ public DatabaseSpecificSQLGenerator(DatabaseTypeResolver databaseTypeResolver) {
     }
 
     public String escape(String arg) {
-        if (databaseTypeResolver.isMySQL()) {
+        return escape(databaseTypeResolver.databaseType(), arg);
+    }
+
+    public static String escape(DatabaseType dialect, String arg) {
```

Comment:
Why do we need to pass the dialect here but not everywhere else?

Reply:
This class is a spring bean, let's not make static methods on it but use it as a proper component.

Reply:
I'd also vote to remove the "static"... we have a lot of "utility" classes that have static methods and are just creating problems down the road (example: just have a look at how a tenant's timezone is resolved). Globally there is also no added value in terms of saving mem or similar (by not instantiating a full object); effectively "@Component"s are singletons... and in that form they are composable... with static they are not... there are plenty of examples where this leads to bloated static functions (more than 500 lines of code). The fix is very easy, the class is already instantiated as a singleton, so there's added value by using the static keyword.

Reply:
... and on top of that: if we don't declare static then we don't have to pass the dialect parameter... and if we don't need to pass that parameter then we don't need to declare a new function which makes this class more explicit and leaves less to guess for the dev.

Reply:
The dialect parameter might not be used for the currently supported dialects but will be needed for the next supported one, that is why this class has a reference to the dialect.
You can not instantiate a component, only from a component (mainly) so if not static then a parameter of the SqlGenerator component for every method in the call chain. That would be another solution to this, right.